src/hotspot/share/gc/shared/objectCountClosure.hpp

```diff
@@ -0,0 +1,29 @@
+#ifndef SHARE_GC_SHARED_OBJECTCOUNTCLOSURE_HPP
+#define SHARE_GC_SHARED_OBJECTCOUNTCLOSURE_HPP
+
+#include "gc/shared/gcId.hpp"
+#include "gc/shared/objectCountEventSender.hpp"
+#include "jfr/jfrEvents.hpp"
+#include "memory/allStatic.hpp"
+#include "runtime/mutex.hpp"
+#include "memory/heapInspection.hpp"
+#include "utilities/macros.hpp"
+#include "utilities/ticks.hpp"
+
+class KlassInfoEntry;
+class Klass;
+
+class ObjectCountClosure : public AllStatic {
```

Comment:
I wouldn't make it a static class.
Extend ObjectClosure instead, which is a StackObj.

(This ties into yr use of static methods below as well.)

We can discuss later this afternoon if the feedback in this review isn't clear.